react/CreditCard.tsx

```diff
@@ -0,0 +1,282 @@
+import React, { useState, useRef, useEffect, useCallback, useMemo } from 'react'
+import { useSSR, useRuntime } from 'vtex.render-runtime'
+import { Button, Spinner } from 'vtex.styleguide'
+import { DocumentField } from 'vtex.document-field'
+import { FormattedMessage, useIntl } from 'react-intl'
+import { useOrderPayment } from 'vtex.order-payment/OrderPayment'
+import { useOrderForm } from 'vtex.order-manager/OrderForm'
+
+import SavedCard from './SavedCard'
+
+let postRobot: any = null
+let iFrameResize: any = null
+if (window?.document) {
+  postRobot = require('post-robot')
+  iFrameResize = require('iframe-resizer/js/iframeResizer')
+}
+
+interface Field {
+  value: string
+  error: boolean
+  errorMessage: string
+  showError: boolean
+}
+
+interface EncryptedCard {
+  encryptedCardNumber: string
+  encryptedCardHolder: string
+  encryptedExpiryDate: string
+  encryptedCsc: string
+}
+
+interface Validator {
+  regex: string
+  mask: string
+  cardCodeRegex: string
+  cardCodeMask: string
+  weights: number[]
+  useExpirationDate: boolean
+  useCardHolderName: boolean
+  useBillingAddress: boolean
+}
+interface PaymentSystem {
+  id: number
+  name: string
+  validator: Validator
+  groupName: string
+}
+
+const IFRAME_APP_VERSION = '0.4.1'
+const LOCAL_IFRAME_DEVELOPMENT = false
```

Comment:
can you read from the querystring to avoid needing to link this app to locally develop the card-form-ui? You could use something like `__localCardUi` as the query string

Reply:
it doesn't seem safe to do that because the user can add that `querystring` and the app be broken.

Reply:
just add a check for `process.env.NODE_ENV !== 'production'`

Reply:
nice